### DISCUSSION TITLE
Very undescriptive and misleading error messaging

### DISCUSSION THREAD

**Discussion by nyck33** [fastapi/fastapi#6592] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#user_routes.py

import fastapi
from models.data.dummy_data import all_coupons
from models.user_models import User

router = fastapi.APIRouter()


@router.get("/get-all-coupons") #response_model=AllCoupons)
async def check_for_new_coupons():
    '''
    todo: make this a get request and just send coupons and check for any new ones in Flutter
    on tap of coupons card in UI
    request payload is current user coupons so check against list of available coupons on server
    send back any new ones eligible for to refresh coupons page on Flutter UI
    :param coupons:
    :return:
    '''
    print(f"responding with list of coupons")
    return all_coupons

@router.post("/user-update-coupons/", response_model=User)
async def update_coupons(user: User):
    '''
    user uses coupon so update the isUsed and return User
    :param user:
    :return: updated User
    '''
    print(user)
    return user

@router.post("/user-delete-coupons", response_model=User)
async def delete_coupons(user:User):
    '''
    when user checks out
    :param user:
    :return:
    '''
    return user

#main.py
import uvicorn
import fastapi
'''
test_json = {"name": "Move out of house", "tasks":[{"description":"get storage",
                                               "complete": "False"},
                                              {"description":"get ispace",
                                                "complete": "False"}]}
'''
from apis import user_routes, other_routes, hotel_routes

api = fastapi.FastAPI
#router = fastapi.APIRouter()

def configure():
    configure_routing()

def configure_routing():
    api.include_router(user_routes.router)
    api.include_router(other_routes.router)
    #apis.include_router(hotel_routes.router)

# for AWS
#https://pypi.org/project/mangum/
#handler = Mangum(app)


if __name__=="__main__":
    #uvicorn.run(app, host="localhost", port=8000)
    #uvicorn.run(app, debug=True, host="0.0.0.0", port=8000)
    configure()
    uvicorn.run(api, port=8000, host='127.0.0.1')
else:
    configure()
```

### Description

TypeError: include_router() missing 1 required positional argument: 'router' was the error for missing () after api = fastapi.FastAPI

That is terrible error messaging.  

### Operating System

Linux

### Operating System Details

Ubuntu LTS 20.04

### FastAPI Version

(serverless) nobu@nobu-IdeaPad-3-15IIL05:/media/nobu/NTFSUbuWin/1_TAP/_Github/coupon_backend$ python -c "import fastapi; print(fastapi.__version__)" 0.68.1

### Python Version

3.9.6

### Additional Context

Dartlang and Flutter also have not so great error messaging.  Python's I've gotten used to somewhat.  But Elm was the gold standard in error messaging and it might be too much to ask but how did it not pick that up on initialization of the app?

**Comment by ghandic** [comment, marked as answer]
This is an implementation issue for the user of the Python code - you need to initialize a class before using it. I suggest you raise an issue with the [psf](https://bugs.python.org) (Python Software Foundation) since this is more of an issue of the language error handling than of FastAPI.

For reference, you would get the same error for this piece of code

```python
class Demo:
    def __init__(self) -> None:
        self.data = ""

    def do_thing(self, router):
        print(router)
        print(self.data)


d = Demo

d.do_thing("xxx")
```

This is because there are two arguments to `do_thing`, `self` - which is the initialised `Demo` object and `router` which is an example of how it is implemented for FastAPI.

**Comment by ghandic** [comment]
PS - A minimal example would look like this

```python
import fastapi
import uvicorn

api = fastapi.FastAPI
router = fastapi.APIRouter()


@router.post("/")
async def example():
    ...


api.include_router(router)

uvicorn.run(api, port=8000, host="127.0.0.1")
```

**Comment by Kludex** [comment]
Everything in python is an object. There's nothing wrong with not instantiating an object from the FastAPI class.

Your issue is in the wrong project. You have a problem with the language, not with FastAPI. 

EDIT: I didn't see the previous replies before writing it 😅

**Comment by nyck33** [comment]
Thanks for your help.  I'll try to help someone now.
